### ISSUE TITLE
How to make relation with UserProfile model?

### ISSUE THREAD

**Issue by webmalinka** [webvimark/user-management#96]
Hello! I have created **UserProfile** model using your instructions. All working fine. Inside model UserProfile I have relation (hasOne) to model User (webvimark\modules\UserManagement\models\User) and I have access to user data from UserProfile model, like **$profileModel->user->username**. But how to set relation inside User model? I want to have access to Profile data, for example, **Yii::$app->user->userProfile->phone**. Should I override original User model, or maybe there is another solution?

**Comment by webvimark** [comment]
The best way is to create some class (let's say UserWithProfile) and extend it from User class. 
In that class you can specify your rules, relations or whatever you want and then call it like this
```php
UserWithProfile::getCurrentUser()->userProfile->phone
```
Or wait some time when I'll add possibility to define your own class in config

**Comment by webmalinka** [comment]
Thanks for reply! I used another solution (maybe temporary): created copy of original User model file (webvimark\modules\UserManagement\models\User) and placed it to the @app/models folder. Then in the application entry script (index.php) I added following string just after require section, to remap User class:
```php
Yii::$classMap['webvimark\modules\UserManagement\models\User'] = '@app/models/User.php';
```
Now in @app/models/User.php I can add whatever I need with saving original relations, behaviors etc. I added relation which I was needed in User class:
```php
public function getUserProfile()
{
    return $this->hasOne(UserProfile::className(), ['user_id' => 'id']);
}
```
I did this remapping, because I couldn't undestand which exactly original user class shoult be extended (there are UserIdentity, UserConfig, User classes in original extension). Besides, if I extend original User class with another class, for example @app/models/MyUser, will I have to rewrite all views and models in extension to use MyUser class properties and content? I needed highly customizable User class - and I've got it:) Works fine for now.
